### PR TITLE
docs: use build/bin/nvim instead of nvim in gen_vimdoc

### PR DESCRIPTION
**Pull request by mfussenegger**
Problem:

`nvim` could point to stable release missing the `nvim -l`
functionality.

Solution:

Require to build nvim first and use `build/bin/nvim`

### CODE REVIEW

**Comment by clason** [comment]
We should also add an explicit error message that the script requires Neovim 0.9 (latest HEAD). I've had it happen that somebody tried to use this with a stable release on their PR and was confused that things didn't work. The error message should also point to setting `$EDITOR` correctly, as I don't think that assumption (like any assumption) holds widely enough.

**Comment by mfussenegger** [comment]
Good point. Updated the message

File: scripts/gen_vimdoc.py [hunk 1]
```diff
@@ -67,7 +67,9 @@
 nvim_version = [line for line in nvim_h_out.split('\n') if '-l ' in line]
 
 if len(nvim_version) == 0:
-    print("\nRequires 'nvim -l' feature, see https://github.com/neovim/neovim/pull/18706")
+    print("\ngen_vimdoc uses `$EDITOR -l` or `nvim -l` if $EDITOR is not set to execute Lua.")
+    print("The '-l' feature is only available in nighty/0.9")
```
Comment (clason) [inline scripts/gen_vimdoc.py:71]:
```suggestion
    print("The '-l' feature is only available in Nvim 0.9")
```

**Comment by mfussenegger** [comment]
Another idea: We could first try `nvim` and if that fails try `../build/bin/nvim` (relative to `gen_vimdoc.py`).

**Comment by clason** [comment]
Or always run `../build/bin/nvim` and give an error that you need to build first if that doesn't exist?

**Pull request merged**